### PR TITLE
More documentation for the debugger

### PR DESCRIPTION
**Pull request by grammati**
I added more details about the debugger commands, explaining what they do.

I'm not actually sure how to preview this locally to see whether it's going to look right in readthedocs, so sorry if I messed up any formatting.

### CODE REVIEW

File: doc/debugging.md [hunk 1]
```diff
@@ -39,26 +39,104 @@ instrumented as well, but they will not be listed by this command.
...
-[Edebug][]. So
-it makes available the following bindings while stepping through code.
+`cider-debug` tries to be consistent with [Edebug][], although there are some
+differences. It makes available the following bindings while stepping through
+code.
 
 Keyboard shortcut               | Description
 --------------------------------|-------------------------------
 <kbd>n</kbd> | Next step
+<kbd>i</kbd> | Step in to a function
```
Comment (Malabarba) [inline doc/debugging.md]: End these in a `.`

Comment (Malabarba) [inline doc/debugging.md]: Sorry. Ignore this.

File: doc/debugging.md [hunk 2]
```diff
@@ -39,26 +39,104 @@ instrumented as well, but they will not be listed by this command.
...
+would not be interesting to stop at, such as constants. For example, there would
+not be much point in stopping execution at a literal number 23 in your code and
+showing that its value is 23 - you already know that.
+
+- **next**: Steps to the next breakpoint
+- **in**: Steps in to the function about to be called. If the next breakpoint is
+  not around a function call, does the same as `next`. Note that not all
+  functions can be stepped in to - only normal functions stored in vars, for
+  which cider can find the source. You cannot currently step in to multimethods,
+  protocol functions, or functions in clojure.core.
```
Comment (Malabarba) [inline doc/debugging.md]: End this sentence with: “(although multimethods and protocols _can_ be instrumented manually)”

File: doc/debugging.md [hunk 3]
```diff
@@ -39,26 +39,104 @@ instrumented as well, but they will not be listed by this command.
...
+  #dbg ^{:break/when (= i 7)}
+  (prn i))
+```
+
+Evaluating the above with `C-M-x`, the debugger will stop only once, when i
+is 7.
+
+You can also have cider insert the break-condition into your code for you. Place
+the point where you want the condition to go and evaluate with `C-u C-u
+C-M-x`.
```
Comment (Malabarba) [inline doc/debugging.md]: (or `C-u C-u C-c C-c`).

**Comment by Malabarba** [comment]
Thanks yet again @grammati.

Other than my localized comments, you'll need to change all ``` `keys` ``` to `<kbd>keys</kbd>`.

Besides that, I think it's good to merge (and very nicely written!).

**Comment by grammati** [comment]
@Malabarba, I made the changes you suggest, thanks for the review.

**Comment by Malabarba** [comment]
@grammati Quick question. Are you on twitter?

**Comment by grammati** [comment]
Sort-of - I'm @perksee on twitter, although I only tweet about twice a year :)

**Comment by Malabarba** [comment]
I just wanted to mention you when tweet about the new debugger stuff.

**Comment by bbatsov** [comment]
> I'm not actually sure how to preview this locally to see whether it's going to look right in readthedocs, so sorry if I messed up any formatting.

There are instructions how to this here http://cider.readthedocs.org/en/latest/about/contributing/#working-on-the-manual

**Pull request merged**